Doc/lane pose calculation (#1192)
tier4/scenario_simulator_v2: docs/developer_guide/lane_pose_calculation/LanePoseCalculation.md

# Lane pose calculation

The calculation method of the lane coordinate system differs from Entity to Entity, and the process of determining which lane to match is complex, including fallback.  
In this document, we will show how the pose calculation of the lane coordinate system is performed for each Entity.

> There are 3 possible timings when the lane coordinate system may be computed for all Entities.  

Reviewer: Duplicate accuracy expressions
```suggestion
There are 3 possible timings when the lane coordinate system is computed for all Entities.  
```

> These are the [timing immediately after the spawn of the Entity](Spawn.md), the [timing of the frame update](UpdateFrame.md) the [timing of the getting longitudinal distance](GetLongitudinalDistance.md)

Reviewer: ```suggestion
These are the [timing immediately after the spawn of the Entity](Spawn.md), the [timing of the frame update](UpdateFrame.md) and the [timing of the getting longitudinal distance](GetLongitudinalDistance.md)
```

!!! Note

>     If you are not a simulator developer but a scenario writer, it is sufficient to check [this document](UpdateFrame.md) and [this section](#lane-coordinate-system-calculation-algorithm-for-a-specific-lane), since the distance measurement via condition in used in the OpenSCENARIO is basically done after frame update.

Reviewer: ```suggestion
    It is sufficient to check [this document](UpdateFrame.md) and [this section](#lane-coordinate-system-calculation-algorithm-for-a-specific-lane), if you are not a simulator developer but a scenario writter. That is because the distance measurement via condition of the OpenSCENARIO is done after frame update basically.
```

| VehicleEntity                      | 2.0               | N/A                                                                                    | 10                                                          |
| PedestrianEntity                   | 2.0               | [2.0 or 4.0](UpdateFrame.md#pedestrian-entity-with-behavior-tree)                      | 10                                                          |
| MiscObject                         | 2.0               | N/A                                                                                    | 10                                                          |

### Detail of the lane coordinate system calculation algorithm for a specific lane

After the lanes to be matched have been determined, the calculation of the specific lane coordinate system to be performed is always based on a per spline curve and line segment determination.

![Lane pose calculation](../../image/lane_pose_calculation.png "Lane pose calculation.")

The centerline of a lane is expressed by the cubic equation [Catmull-Rom spline.](https://en.wikipedia.org/wiki/Centripetal_Catmull%E2%80%93Rom_spline)  

> Catmull-Rom spline and line perturbation are implemented by finding the multiple solution of one cubic equation.  

Reviewer: Does `line perturbation` have the same meaning as `line segment` elesewhere?
If they mean the same thing, please use the same expression.
(Also, the expressions used in the figure and text are different, making it difficult to understand...)

Author: I fixed to `line segment` in https://github.com/tier4/scenario_simulator_v2/pull/1192/commits/929775a6c2974f8ce9229bd9b5a215249beced5b
